vision/medical_imaging/3d-unet-kits19/README.md

```diff
@@ -0,0 +1,73 @@
+  - `backend` choices: `pytorch`, `onnxruntime`, `tensorflow`
+  - `test` choices: `accuracy`, `performance`
+
+Other useful commands:
+- `make preprocess_calibration_data`: preprocess calibration dataset
+- `make convert_onnx_model`: convert pytorch model to ONNX
+- `make convert_tensorflow_model`: convert ONNX model to TensorFlow
+- `make download_kits19_raw_data`: manually download KITS19 raw data
+- `make download_models`: manually download all the models (pytorch, ONNX, tensorflow)
+- `python3 run.py --backend=[tensorflow|pytorch|onnxruntime] --scenario=[Offline|SingleStream|MultiStream|Server] [--accuracy] --model=[path/to/model]`: run the harness inside the docker container. Performance or Accuracy results will be printed in console
+- `python3 accuracy_kits.py --log_file=<LOADGEN_LOG>`: compute accuracy from a LoadGen accuracy JSON log file
+
+## Details
+
+- Baseline SUT that connects LoadGen and QSL is in [base_SUT.py](base_SUT.py). QSL implementation is in [kits_QSL.py](kits_QSL.py).
+- SUT implementations for the above supported backends are in [pytorch_SUT.py](pytorch_SUT.py), [onnxruntime_SUT.py](onnxruntime_SUT.py), and [tensorflow_SUT.py](tensorflow_SUT.py).
+- The script [accuracy_kits.py](accuracy_kits.py) parses LoadGen accuracy log, post-processes it, and computes the accuracy.
+- Preprocessing and evaluation (including post-processing) are not included in the timed path.
+- Unlike BraTS19 data, KiTS19 data come in various shapes with various voxel spacing. In this app, we preprocess images and labels so that they are under same voxel spacing and ready for sub-volume inferencing.
+- When preprocessing is finished, reshaped imaging/segmentation NIfTI files are also populated under preprocessed data directory, specifically under `nifti` sub-directory.
+- When postprocessing is finished, either as part of `make run_[backend]_accuracy` or `python3 accuracy_kits.py`, prediction results, i.e. segmentations done by inference, are populated in the postprocess data directory, together with `summary.csv` file showing DICE scores of Kidney segmentation and Tumor segmentation.
+- User can view these NIfTI files with NIfTI viewers like [Mango](http://ric.uthscsa.edu/mango/) as below:
+  - Image, Segmentation:ground-truth, Segmentation:prediction
+    ![Image, Segmentation:ground-truth, Segmentation:prediction](pics/img_segm_pred.png)
+  - Overlaid Image with Segmentation:ground-truth, Overlaid Image with Segmentation:prediction
+    ![Overlaid Image with Segmentation:ground-truth, Overlaid Image with Segmentation:prediction](pics/overlay_segm_pred.png)
+
+## Calibration Set
+
+The calibration set is tentatively determined to have 20 images listed in [Calibration set](meta/calibration_cases.json). They are randomly selected from the training set used in [MLPerf-Training 3D-UNet](https://github.com/mlcommons/training/blob/master/image_segmentation/pytorch). None of the image in the calibration set is included in the inference set.
```

Comment:
What is the intended calibration method that supposed to work? (min-max, per tensor, etc. which works for BRATS19).   It might bring peace of mind if certain calibration method are known to work.

Reply:
At the moment, Per-Tensor, Min/Max or MSE would be the good candidates.